reference/orders.v2.oas2.yml

```diff
@@ -4412,13 +4620,229 @@ components:
         - type: object
           properties:
             products:
-              type: array
-              items:
-                anyOf:
-                  - $ref: '#/components/schemas/orderCustomProduct_Post'
-                  - $ref: '#/components/schemas/orderCatalogProduct_Post'
+              $ref: '#/components/schemas/orderProduct_Post'
             shipping_addresses:
               type: array
               items:
                 $ref: '#/components/schemas/shippingAddress_Base'
+            consignments:
+              type: array
+              minItems: 1
+              maxItems: 1
+              description: |-
+                A list of order consignments.
+
+                Consignments field can not be used in conjunction with the products or the shipping_addresses fields.
+
+                Currently only single consignment orders are supported.
+              items:
+                $ref: '#/components/schemas/orderConsignment_Post'
         - $ref: '#/components/schemas/order_Shared'
+    orderConsignment_Post:
+      title: orderConsignment_Post
+      type: object
+      properties:
+        line_items:
+          $ref: '#/components/schemas/orderProduct_Post'
+        fulfillment_method:
+          allOf:
+            - $ref: '#/components/schemas/orderFulfillmentMethod'
+            - type: object
+              properties:
+                pickup:
+                  type: object
+                  description: ''
+                  properties:
+                    location:
+                      type: object
+                      description: |-
+                        - The default values for optional fields come from the pickup store location specified via id field. Use the optional fields for the values you need to overwrite.
+
+                        - If inventory is being tracked on an item, there should be sufficient stock for the item in the pickup location, and the related inventory will be deducted upon successful creation.
+                      properties:
+                        id:
+                          type: integer
+                      required:
+                        - id
```

Comment:
This seems to be duplicates of orderPickup, should we merge the "description" and "required" info to orderPickup model ?

Reply:
My intention was to distinguish between required fields and description of POST and PUT:

location id is not required for PUT and the  description for PUT is:

- If pickup location id is updated, rest of the location fields will be updated to reflect the new location information. If you need to overwrite these values you should provide them in the payload.
                            
- If inventory is being tracked on an item, there should be sufficient stock for the item in the new pickup location, and the related inventory will be deducted from the new location and return to the old location upon successful update.

Reply:
ah I see, TIL about that trick, it' really smart 😄. Thanks for explanation Dorna 🙇

Reply:
Haha, nw 🙇‍♀️
It's a bit messy in the Yaml format, but the rendered result is good.